effekt/shared/src/main/scala/effekt/core/Parser.scala

```diff
@@ -168,13 +174,19 @@ class CoreParsers(positions: Positions, names: Names) extends EffektLexers(posit
 
   // Blocks
   // ------
-  lazy val block: P[Block] =
+  lazy val block: P[Block] = {
+    ( blockNonMember ~ many((`.` ~> id) ~ (`:` ~> blockType)) ^^ {
+      case firstReceiver ~ fields => fields.foldLeft(firstReceiver) {
+        case (receiver, field ~ tpe) => Block.Member(receiver, field, tpe)
+      }
+    }) | blockNonMember
+  }
```

Comment:
typically we would drop the braces around the body and put `| blockNonMember` on a new line to read like a grammar in a paper :)